Source/Charts/Renderers/YAxisRenderer.swift

```diff
@@ -387,6 +387,52 @@ open class YAxisRenderer: AxisRendererBase
             }
         }
         
+        renderLimitLineFill(context: context)
+        
         context.restoreGState()
     }
+    
+    open func renderLimitLineFill(context : CGContext) {
+        
+        guard
+            let yAxis = self.axis as? YAxis,
+            let transformer = self.transformer
+            else { return }
+        
+        if !yAxis.isEnabled || !yAxis.isLimitLineFillEnabled
+        {
+            return
+        }
+        
+        var limitLines = yAxis.limitLines
+        
+        if limitLines.count != 2
+        {
+            return
+        }
+        
+        context.saveGState()
```

Comment:
you never call `context.restoreGState()`. Please add it in a single line `defer` statement, i.e 
```
defer { context.restoreGState() }
```

Reply:
Relying on the call-site to call half of the necessary setup/teardown isn't good practice. Either the caller needs to do both the setup and the tear down, or the method should handle it itself. In this case, the method should handle it. Please add the `defer` statement.